pyproject.toml

```diff
@@ -27,13 +27,10 @@ packages = [
 "Bug Tracker" = "https://github.com/argoproj-labs/hera-workflows/issues"
 
 [tool.poetry.dependencies]
-python = ">=3.7, <3.11"
-pydantic = "^1.9.0"
-python-dateutil = "^2.8.2"
```

Comment:
I am trying to recall why this was introduced... I have a hunch it has to do with the way deserialization of cron workflow payloads occurs, which might have a runtime dependency on `python-dateutil` because it assigns a datetime stamp to a field that indicates the last time a cron workflow ran. This happens even when the cron workflow is created at first. However, I could be wrong. Any chance you can submit one of the examples to an Argo cluster and see whether workflows create successfully with this environment configuration?

Reply:
Yup, I can test that out in the morning!

I saw we use pytz in the cron module to confirm timezones (I think those are now also in the stdlib, but only for 3.9+ or something), but I didn't see dateutil used directly. It is still brought in as a transitive dependency via `argo-workflows` however, so should still be available. The only dep that got removed completely was `certifi`.

Reply:
Looking good, `python-dateutil` still got installed transitively and the scheduled job created alright!

<img width="1193" alt="Screen Shot 2022-08-11 at 09 58 27" src="https://user-images.githubusercontent.com/2555532/184151003-e19eb0c6-eaf5-4b0b-8116-5bcd6507d686.png">

<details>
<summary>Log</summary>

```sh
[I] ~/hera-test $ python3 -m venv .venv
[I] ~/hera-test $ source .venv/bin/activate.fish
[I] (.venv) ~/hera-test $ pip3 install -e /Users/jacobhayes/src/github.com/argoproj-labs/hera-workflows
Obtaining file:///Users/jacobhayes/src/github.com/argoproj-labs/hera-workflows
  Installing build dependencies ... done
  Checking if build backend supports build_editable ... done
  Getting requirements to build editable ... done
  Preparing editable metadata (pyproject.toml) ... done
Collecting argo-workflows==6.3.5
  Using cached argo_workflows-6.3.5-py3-none-any.whl (1.3 MB)
Collecting pydantic<2.0.0,>=1.9.0
  Downloading pydantic-1.9.2-cp39-cp39-macosx_11_0_arm64.whl (2.5 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 2.5/2.5 MB 27.2 MB/s eta 0:00:00
Collecting pytz
  Using cached pytz-2022.1-py2.py3-none-any.whl (503 kB)
Collecting urllib3>=1.25.3
  Using cached urllib3-1.26.11-py2.py3-none-any.whl (139 kB)
Collecting python-dateutil
  Using cached python_dateutil-2.8.2-py2.py3-none-any.whl (247 kB)
Collecting typing-extensions>=3.7.4.3
  Using cached typing_extensions-4.3.0-py3-none-any.whl (25 kB)
Collecting six>=1.5
  Using cached six-1.16.0-py2.py3-none-any.whl (11 kB)
Building wheels for collected packages: hera-workflows
  Building editable for hera-workflows (pyproject.toml) ... done
  Created wheel for hera-workflows: filename=hera_workflows-3.6.3-py3-none-any.whl size=5399 sha256=df3e2f92e2727fb708c0b9a465e479fb7e281d91436cd7ed4c0b8df485886133
  Stored in directory: /private/var/folders/ss/tgmzwbx52y35qpr7rzrtv8ym0000gn/T/pip-ephem-wheel-cache-4nwz6v81/wheels/02/7f/31/bd06123ae4a7affe003677ed2df75772f5694403c190eb6025
Successfully built hera-workflows
Installing collected packages: pytz, urllib3, typing-extensions, six, python-dateutil, pydantic, argo-workflows, hera-workflows
Successfully installed argo-workflows-6.3.5 hera-workflows-3.6.3 pydantic-1.9.2 python-dateutil-2.8.2 pytz-2022.1 six-1.16.0 typing-extensions-4.3.0 urllib3-1.26.11
WARNING: There was an error checking the latest version of pip.
[I] (.venv) ~/hera-test $ cat cron.py
"""This example showcases the cron hello world example of Hera"""

from hera import CronWorkflow, CronWorkflowService, Task


def hello():
    print("Hello, Hera!")


with CronWorkflow(
    "hello-hera-cron",
    "5 4 * * *",
    service=CronWorkflowService(
        host="redacted",
        token="redacted",
    ),
    timezone="UTC",
) as cw:
    Task("t", hello)

cw.create()
[I] (.venv) ~/hera-test $ python3 cron.py
[I] (.venv) ~/hera-test $
```

</details>